tests/_helpers/test-runner.js

```diff
@@ -0,0 +1,42 @@
+const chalk = require( 'chalk' );
+const sendFiles = require( './github-files' );
+const { dispatchWorkflow, verifyWorkflowStatus, getRunningActions } = require( './github-workflows' );
+
+async function runTests( tests ) {
+	const testCase = tests.shift();
+
+	if( testCase ) {
+		console.log( '\n *** Running test for: ' + chalk.blue( testCase.name ) + ' ***' );
+		await runTest( testCase );
+		return runTests( tests );
+	}
+}
+
+async function runTest( testCase ) {
+	return new Promise( async ( resolve, reject ) => {
```

Comment:
I have the same proposal for getting rid of `new Promise` as for the `verifyWorkflowStatus()`. WDYT?